[Refact] #45 clipboard 라이브러리 제거

sewonkimm opened the pull request.

sewonkimm committed:
[Refact] #44 jquery 라이브러리 제거

sewonkimm committed:
Merge pull request #47 from mojimojicorp/Refact/#44

[Refact] #44 jquery 라이브러리 제거

soyoungjeong committed:
[Refact] #18 web worker로 로딩 속도 개선

innerHTML 해줄 string을 한 번에 구하다 보니 연산 속도가 길어져
카테고리별로 나누어 web worker로 멀티스레드로 연산하도록 수정.

sewonkimm committed:
[Refact] #45 clipboard 라이브러리 제거

- clipboard API로 충분히 구현이 가능하므로 기존의 clipboard 라이브러리를 제거했습니다.
- copy 버튼을 클릭하면 clipboard에 복사되도록 했습니다.

soyoungjeong committed:
[Refact] #18 worker 역할 세분화

people의 개수가 많아 느리게 뜨는 현상 때문에 people을 세분화해서
worker의 역할 세분화.

soyoungjeong committed:
[React] #18 이모지 데이터 저장 구조 변경

sewonkimm committed:
Merge branch 'Refact/#18' into Refact/18

sewonkimm committed:
Merge pull request #48 from mojimojicorp/Refact/18

Refact/18 로딩 속도 개선

sewonkimm committed:
Merge Refact/#18

sewonkimm committed:
[Chore] merge 후 작동하지 않는 기능 동작하도록 수정

- Header의 panel, setting 동작하도록 수정
- 기존 panel 코드의 jquery 제거
- search 기능 동작하도록 수정

sewonkimm committed:
Merge

sewonkimm committed:
Merge

sewonkimm committed:
[Refact] #45 emoji 렌더링 비동기 처리

각 emoji 버튼에 클릭이벤트를 달아주기 위해서 attachEmoji 작업이 다
끝나고 이벤트리스너를 달아주도록 작업하였습니다.

sewonkimm committed:
[Chore] setGrid 순서 조정

sewonkimm committed:
[Refact] #45 copy setting에 따라서 copy 기능 구현

soyoungjeong commented on js/copy/setCopied.mjs:
```diff
@@ -0,0 +1,14 @@
+  }, 200);
+}
+
+export { setCopied };
```
copy가 완료되었을 때 버튼의 동작이니까 setCopied 보단 setCopyCompleteButton 등이 괜찮지 않을까요??

sewonkimm replied:
의미가 더 명확해지고 좋네요! 수정하겠습니다~

sewonkimm replied:
버튼이아니라 함수니까 이름을 setCopyButtonComplete로 하는 게 더 나을 것 같아요.

soyoungjeong replied:
아니면 setCopiedButton도 좋은 것 같아요!!

soyoungjeong commented on js/emoji/attachEmoji.mjs:
```diff
@@ -1,79 +1,77 @@
+  count++;
+  if (count >= 10) {
+    res();
+  }
```
그냥 질문입니다ㅎㅎ count는 왜 세는 것인가요???

sewonkimm replied:
웹 워커 로딩이 다 끝나기 전에 이벤트 리스너를 붙인는 함수가 실행되면 어떤 이모티콘에는 이벤트 리스너가 붙지 않는 현상이 발생합니다. 그래서 웹 워커 로딩이 다 끝날 때 resolve()를 해주기 위해 count를 세도록 만들었습니다!

soyoungjeong commented on js/emoji/getEmoji.js:
```diff
@@ -1,40 +1,33 @@
-// } else if (copySetting === 'manual') {
-//   onecopy();
-// }
+getEmoji();
```
이곳에서 함수를 호출하는 것보다 afterLoad에서 Util.includeJS('../js/emoji/getEmoji.js'); 이 부분 대신 getEmoji()로 호출하는 것이 더 자연스러워 보입니다~!

sewonkimm replied:
그렇네요. 수정 반영하겠습니다~

sewonkimm committed:
[Chore] Review 반영

1. setCopied를 SetCopyButtonComplete으로 함수명 변경하며 의미가 더 명확해 지도록 함.
2. getEmoji 함수 호출 수정

soyoungjeong reviewed (approved):
setCopiedButton/setCopyButtonComplete 중에 마음에 드시는 것으로 하시고 merge하시면 될 것 같아요! 수고하셨습니닷😀

sewonkimm committed:
[Fix] #45 copy setting 변경 시 copy 기능 변경 안되는 버그 수정

- setCopyEvent에서 copy버튼, reset버튼에 이벤트리스너 등록하는 코드를 분리했습니다.
- 이모지에 이벤트리스너를 등록할 때, 기존에 등록되어있는 이벤트리스너를 제거 후 설정에 맞는 이벤트리스너가 등록되어 설정을 변경하면 바로 변경된 copy 기능이 반영될 수 있도록 했습니다.
- removeEventListener 메서드 사용을 위해 이벤트리스너를 함수로 묶었습니다.
- git-commit-template 파일 추가

sewonkimm committed:
[Chore] 확장앱 실행을 위한 recent 기능 주석처리

soyoungjeong reviewed (approved):
LGTM : )
